rust/otap-dataflow/crates/otap/src/experimental/gigla_exporter/exporter.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use opentelemetry_proto::tonic::collector::logs::v1::ExportLogsServiceRequest;
+use otap_df_engine::control::NodeControlMsg;
+use otap_df_engine::error::Error;
+use otap_df_engine::local::exporter::{EffectHandler, Exporter};
+use otap_df_engine::message::{Message, MessageChannel};
+use otap_df_engine::terminal_state::TerminalState;
+use otap_df_pdata::otlp::OtlpProtoBytes;
+use prost::Message as _;
+
+use crate::experimental::gigla_exporter::config::Config;
+use crate::experimental::gigla_exporter::gigla_client::GigLaClient;
+use crate::experimental::gigla_exporter::transformer::Transformer;
+use crate::pdata::OtapPdata;
+
+/// GigLA exporter sending telemetry to the GigLA backend.
+///
+/// This exporter processes OTLP logs and sends them to Azure GigLA
+/// (Geneva Infrastructure General-purpose Logging Analytics).
+pub struct GigLaExporter {
+    config: Config,
+    client: GigLaClient,
+    transformer: Transformer,
+}
+
+impl GigLaExporter {
+    /// Build a new exporter from configuration.
+    pub fn new(config: Config) -> Result<Self, otap_df_config::error::Error> {
+        // Validate configuration
+        config
+            .validate()
+            .map_err(|e| otap_df_config::error::Error::InvalidUserConfig { error: e })?;
+
+        // Create GigLA client with the full config
+        let client = GigLaClient::new(&config)
+            .map_err(|e| otap_df_config::error::Error::InvalidUserConfig { error: e })?;
+
+        // Create log transformer
+        let transformer = Transformer::new(&config);
+
+        Ok(Self {
+            config,
+            client,
+            transformer,
+        })
+    }
+
+    /// Handle a single pdata message.
+    async fn handle_pdata(
+        &self,
+        pdata: OtapPdata,
+        effect_handler: &EffectHandler<OtapPdata>,
+    ) -> Result<(), String> {
+        // Split pdata into context and payload
+        let (_context, payload) = pdata.into_parts();
+
+        // Convert OTAP payload to OTLP bytes
+        // TODO: This conversion step should be eliminated (see method documentation above)
```

Comment:
What's this about? I couldn't find what this is referring to.

Reply:
https://github.com/open-telemetry/otel-arrow/issues/1441

cc @lalitb